tests/agent_server/deploy/e2e/test_facts_new.py

```diff
@@ -0,0 +1,439 @@
+
+    await clienthelper.put_version_simple(resources, version)
+
+    result = await client.release_version(env_id, version, True, const.AgentTriggerMethod.push_full_deploy)
+    assert result.code == 200
+
+    result = await client.get_param(env_id, "length", resource_id_wov)
+    assert result.code == 503
+
+    env_uuid = uuid.UUID(env_id)
+
+    async def has_params():
+        params = await data.Parameter.get_list(environment=env_uuid, resource_id=resource_id_wov)
+        return len(params) >= 3
+
+    await retry_limited(has_params, 5)
+
+    result = await client.get_param(env_id, "key1", resource_id_wov)
+    assert result.code == 200
+    no_error_in_logs(caplog)
+
+
+async def test_purged_facts(resource_container, client, clienthelper, agent, environment, no_agent_backoff, caplog):
+    """
+    Test if facts are purged when the resource is purged.
+    """
+    resource_container.Provider.set("agent1", "key", "value")
+
+    version = await clienthelper.get_version()
+    resource_id_wov = "test::Resource[agent1,key=key]"
+    resource_id = "%s,v=%d" % (resource_id_wov, version)
+
+    resources = [{"key": "key", "value": "value", "id": resource_id, "requires": [], "purged": False, "send_event": False}]
+
+    await clienthelper.put_version_simple(resources, version)
+
+    result = await client.release_version(environment, version, True, const.AgentTriggerMethod.push_full_deploy)
+    assert result.code == 200
+
+    result = await client.get_param(environment, "length", resource_id_wov)
+    assert result.code == 503
+
+    env_uuid = uuid.UUID(environment)
+
+    async def wait_for_three_params():
+        params = await data.Parameter.get_list(environment=env_uuid, resource_id=resource_id_wov)
+        return len(params) >= 3
+
+    await retry_limited(wait_for_three_params, 10)
+
+    result = await client.get_param(environment, "key1", resource_id_wov)
+    assert result.code == 200
+
+    # Purge the resource
+    version = await clienthelper.get_version()
+    resources[0]["id"] = "%s,v=%d" % (resource_id_wov, version)
+    resources[0]["purged"] = True
+    result = await client.put_version(
+        tid=environment,
+        version=version,
+        resources=resources,
+        unknowns=[],
+        version_info={},
+        compiler_version=get_compiler_version(),
+    )
+    assert result.code == 200
+    result = await client.release_version(environment, version, True, const.AgentTriggerMethod.push_full_deploy)
+    assert result.code == 200
+
+    result = await client.get_version(environment, version)
+    assert result.code == 200
+
+    await _wait_until_deployment_finishes(client, environment, version)
+
+    result = await client.get_version(environment, version)
+    assert result.result["model"]["done"] == len(resources)
+
+    # The resource facts should be purged
+    result = await client.get_param(environment, "length", resource_id_wov)
+    assert result.code == 503
```

Comment:
I'm not sure to understand how this proves that the fact has been purged, I only see this in the result:
```
'Agents queried for resource parameter.'
```

Reply:
I copied pasted this testcase to ensure the behavior is the same.

This at least shows it is dropped from the DB (it was 200, it hasn't timed out, now it is gone,...)